module/optee/console/src/mod_optee_console.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Arm SCP/MCP Software
+ * Copyright (c) 2022, Linaro Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Description:
+ *     Interface SCP-firmware console module with OP-TEE console resources.
+ */
+
+#include <fwk_assert.h>
+#include <fwk_attributes.h>
+#include <fwk_event.h>
+#include <fwk_log.h>
+#include <fwk_mm.h>
+#include <fwk_module.h>
+#include <fwk_module_idx.h>
+#include <fwk_status.h>
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <console.h>
+
+static int mod_console_init(
+    fwk_id_t module_id,
+    unsigned int element_count,
+    const void *data)
+{
+    return FWK_SUCCESS;
+}
+
+static int mod_console_element_init(
+    fwk_id_t element_id,
+    unsigned int unused,
+    const void *data)
+{
+    return FWK_SUCCESS;
+}
+
+
+static int mod_console_io_open(const struct fwk_io_stream *stream)
+{
+    return FWK_SUCCESS;
+}
+
+static int mod_console_io_getch(
+    const struct fwk_io_stream *restrict stream,
+    char *restrict ch)
+{
+    *ch =  'A';
+
+    return FWK_SUCCESS;
+}
+
+static int mod_console_io_putch(const struct fwk_io_stream *stream, char ch)
+{
+    console_putc(ch);
```

Comment:
where is this created?
I don't know if it is helpful but `stdio` module integrates `fprintf` functions from stdio library

Reply:
console_putc() is the optee-os function to output on the optee console